common.go

```diff
@@ -76,6 +81,33 @@ const KillRingMax = 60
 // HistoryLimit is the maximum number of entries saved in the scrollback history.
 const HistoryLimit = 1000
 
+// NewLiner initializes a new *State, and sets the terminal into raw mode. To
+// restore the terminal to its previous state, call State.Close().
+func NewLiner() *State {
+	var s State
+	s.setWriter(os.Stdout)
+	s.setReader(os.Stdin)
+	s.init()
+	return &s
+}
+
+// NewLinerTTY initializes a new *State with connecting to tty.
+// This is useful when prompting regardless of the redirections.
+func NewLinerTTY() *State {
```

Comment:
I'm confused. Isn't the entire point to allow the caller to choose which TTY they want liner to use? Shouldn't the tty be passed in here, instead of calling tty.Open()?